homeassistant/components/backup/manager.py

```diff
@@ -1364,37 +1370,84 @@ def _update_issue_backup_failed(self) -> None:
-        if not agent_errors and not unavailable_agents:
+
+        addon_errors = written_backup.addon_errors
+        folder_errors = written_backup.folder_errors
+
+        if (
+            not agent_errors
+            and not unavailable_agents
+            and not addon_errors
+            and not folder_errors
+        ):
+            # No issues to report, clear previous error
             ir.async_delete_issue(self.hass, DOMAIN, "automatic_backup_failed")
             return
-        ir.async_create_issue(
-            self.hass,
-            DOMAIN,
-            "automatic_backup_failed",
-            is_fixable=False,
-            is_persistent=True,
-            learn_more_url="homeassistant://config/backup",
-            severity=ir.IssueSeverity.WARNING,
-            translation_key="automatic_backup_failed_upload_agents",
-            translation_placeholders={
-                "failed_agents": ", ".join(
-                    chain(
-                        (
-                            self.backup_agents[agent_id].name
-                            for agent_id in agent_errors
-                        ),
-                        unavailable_agents,
+        if (agent_errors or unavailable_agents) and not (addon_errors or folder_errors):
+            # No issues with add-ons or folders, but issues with agents
+            self._create_automatic_backup_failed_issue(
+                "automatic_backup_failed_upload_agents",
+                {
+                    "failed_agents": ", ".join(
+                        chain(
+                            (
+                                self.backup_agents[agent_id].name
+                                for agent_id in agent_errors
+                            ),
+                            unavailable_agents,
+                        )
                     )
-                )
-            },
-        )
+                },
+            )
+        elif addon_errors and not (agent_errors or unavailable_agents or folder_errors):
+            # No issues with agents or folders, but issues with add-ons
+            self._create_automatic_backup_failed_issue(
+                "automatic_backup_failed_addons",
+                {"failed_addons": ", ".join(addon for addon in addon_errors)},
+            )
+        elif folder_errors and not (agent_errors or unavailable_agents or addon_errors):
+            # No issues with agents or add-ons, but issues with folders
+            self._create_automatic_backup_failed_issue(
+                "automatic_backup_failed_folders",
+                {"failed_folders": ", ".join(folder for folder in folder_errors)},
+            )
+        else:
+            # Issues with agents, add-ons, and/or folders
+            self._create_automatic_backup_failed_issue(
+                "automatic_backup_failed_agents_addons_folders",
+                {
+                    "failed_agents": ", ".join(
```

Comment:
Please use a parenthesis and indent the item, if it spans more then one line, so it's clear where it starts and ends. Currently the line breaks and the next item have the same indentation.